Yoast/javascript
# Also stem derivational suffixes at start of stemming

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Makes sure that words are stemmed correctly when they are on the `doNotStemParticle` or `doNotStemPronounSuffixes` exception lists. This concerns instances of these words that occur with derivational suffixes. For example, `terhimpun` gets stemmed to`himpun`. 
* [Yoast SEO Premium] Improves the stemming of words that look like they end in valid non-derivational suffixes -ku/-mu/-nya/-kah/-lah/-pun


## Relevant technical choices:

* Add derivational morphology stemming step at the beginning only for the purpose of checking whether the word (with derivational morphology stemmed) is on the exception list and otherwise discard this stemmed form.

## Test instructions

This PR can be tested by following these steps:

Check the unit tests:
* Run yarn test and make sure all tests still pass.
* Add more word forms of words that look like they end in valid non-derivational suffixes -ku/-mu/-nya/-kah/-lah/-pun which get derivational affixes (i.e. meng-/peng-/men-/pen-/mem-/pem-/me-/di-/ter-/ke-/ber-/be-/per-/pe-). Example: mengalah -> kalah, bercelah -> celah, pemecahbelah -> pecahbelah, dijumlah -> jumlah, menjumlah -> jumlah, merekah -> rekah.

Check in the content-analysis app:

* Open the content analysis app by running yarn start;
* Insert a text of al least 150 words;
* Set the locale to Indonesian (i.e., id_ID);
* Make sure you have different forms of words that look like they end in valid non-derivational suffixes which get derivational affixes you want to test in the text and set the word
you want to test as a keyword (e.g., penjumlah, bersekolah, menikah );
* Check on the eye icon that all the word forms become highlighted (e.g. menikah, other forms: nikah, nikahan, menikahlah, pernikahan, menikahi, dinikahi)

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-388
